src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -486,19 +508,24 @@ end = struct
     in
     go (empty_hash_at_height start_height) hash start_height
 
+  (** Given an address and the hash of the corresponding subtree, start getting
+      the children.
+  *)
   let handle_node t addr exp_hash =
-    if Addr.depth addr >= MT.depth - subtree_height then (
+    if Addr.depth addr >= MT.depth - account_subtree_height then (
       expect_content t addr exp_hash ;
-      Linear_pipe.write_without_pushback_if_open t.queries
+      Linear_pipe.write_without_pushback t.queries
```

Comment:
these should still be a `_if_open`, I believe. @johnwu93 recently changed this because he was seeing the query pipe being closed while some jobs were still outstanding.

Reply:
Yes, please change back to `write_without_pushback_if_open`